Mathlib/RingTheory/Valuation/AlgebraInstances.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández, Filippo A. E. Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández, Filippo A. E. Nuccio
+-/
+import Mathlib.Algebra.Order.Group.TypeTags
+import Mathlib.RingTheory.IntegralClosure.IsIntegralClosure.Basic
+import Mathlib.RingTheory.Valuation.ValuationSubring
+
+/-!
+# Algebra instances
+
+This file contains several `Algebra` and `IsScalarTower` instances related to extensions
+of a field with a valuation, as well as their unit balls.
+
+# Main Definitions
+* `ValuationSubring.algebra` : Given an algebra between two field extensions `L` and `E` of a
+  field `K` with a valuation, create an algebra between their two rings of integers.
+
+# Main Results
+
+* `integralClosure_algebraMap_injective` : the unit ball of a field `K` with respect to a
+  valuation injects into its integral closure in a field extension `L` of `K`.
+-/
+
+open Function Valuation
+
+open scoped DiscreteValuation
+
+variable {K : Type*} [Field K] (v : Valuation K ℤₘ₀) (L : Type*) [Field L] [Algebra K L]
+
+namespace ValuationSubring
+
+-- Implementation note : this instance was automatic in Lean3
+instance smul : SMul v.valuationSubring (integralClosure v.valuationSubring L) := Algebra.toSMul
+
+theorem algebraMap_injective : Injective (algebraMap v.valuationSubring L) :=
+  (NoZeroSMulDivisors.algebraMap_injective K L).comp (IsFractionRing.injective _ _)
+
+theorem isIntegral_of_mem_ringOfIntegers {x : L} (hx : x ∈ integralClosure v.valuationSubring L) :
+    IsIntegral v.valuationSubring (⟨x, hx⟩ : integralClosure v.valuationSubring L) := by
```

Comment:
Could you please provide a variant for bundled elements `x : integralClosure v.valuationSubring L`?

Reply:
Done